blackbox/docs/administration/ssl.txt

```diff
@@ -114,3 +114,272 @@ string. E.g.::
+
+Command::
+
+    openssl x509 -req -in server.csr -CA rootCA.crt -CAkey rootCA.key -CAcreateserial -out server.crt -sha256 -days 36500
+
+Output::
+
+    Signature ok
+    subject=/C=AT/ST=Vorarlberg/L=Dornbirn/O=crate.io/OU=PSQL Netty SSL/CN=CrateDB
+    Getting CA Private Key
+
+
+Optional: Import CA into Keystore
+---------------------------------
+
+The CA needs to be imported to the Keystore for the certificate chain to be
+available when we import our signed certificate.
+
+Command::
+
+    keytool -import -keystore keystore -file rootCA.crt -alias theCARoot
+
+Output::
+
+    Enter keystore password:
+    Owner: EMAILADDRESS=info@crate.io, CN=crate.io, OU=Cryptography Department, O=Crate.io, L=Dornbirn, ST=Vorarlberg, C=AT
+    Issuer: EMAILADDRESS=info@crate.io, CN=crate.io, OU=Cryptography Department, O=Crate.io, L=Dornbirn, ST=Vorarlberg, C=AT
+    Serial number: f13562ec6184401e
+    Valid from: Mon Jun 12 13:09:17 CEST 2017 until: Wed May 19 13:09:17 CEST 2117
+    Certificate fingerprints:
+         MD5:  BB:A1:79:53:FE:71:EC:61:2A:19:81:E8:0E:E8:C9:81
+         SHA1: 96:66:C1:01:49:17:D1:19:FB:DB:83:86:50:3D:3D:AD:DA:F7:C6:A9
+         SHA256: 69:82:C5:24:9A:A1:AE:DF:80:29:7A:26:92:C1:A5:9F:AF:7D:03:56:CC:C3:E9:73:3B:FD:85:66:35:D6:8A:9B
+         Signature algorithm name: SHA256withRSA
+         Version: 3
+
+    Extensions:
+
+    #1: ObjectId: 2.5.29.35 Criticality=false
+    AuthorityKeyIdentifier [
+    KeyIdentifier [
+    0000: CD 29 4E 07 3D C3 7C D0   16 45 FB 0A CE 8D B4 98  .)N.=....E......
+    0010: B7 A8 4C 79                                        ..Ly
+    ]
+    [EMAILADDRESS=info@crate.io, CN=crate.io, OU=Cryptography Department, O=Crate.io, L=Dornbirn, ST=Vorarlberg, C=AT]
+    SerialNumber: [    f13562ec 6184401e]
+    ]
+
+    #2: ObjectId: 2.5.29.19 Criticality=false
+    BasicConstraints:[
+      CA:true
+      PathLen:2147483647
+    ]
+
+    #3: ObjectId: 2.5.29.14 Criticality=false
+    SubjectKeyIdentifier [
+    KeyIdentifier [
+    0000: CD 29 4E 07 3D C3 7C D0   16 45 FB 0A CE 8D B4 98  .)N.=....E......
+    0010: B7 A8 4C 79                                        ..Ly
+    ]
+    ]
+
+    Trust this certificate? [no]:  yes
+    Certificate was added to keystore
+
+
+Import Signed Certificate
+-------------------------
+
+Now we have a signed certificate, signed by either from the offical CA
+or our own CA. Let's import it to the keystore.
+
+Command::
+
+    keytool -import -keystore keystore -file server.crt -alias server
+
+Output::
+
+    Enter keystore password:
+    Enter key password for <server>
+    Certificate reply was installed in keystore
+
+
+Optional: Import CA into Truststore
+-----------------------------------
+
+If we are using our own CA, we should also import the certificate to the
+Truststore, such that it is available for clients which want to verify
+signatures.
+
+Command::
+
+    keytool -import -keystore truststore -file rootCA.crt -alias theCARoot
+
+Output::
```

Comment:
I would avoid printing the console outputs of `keytool` here in the docs since they have no user value. I would rather mention what the command does and highlight the outcoming `.crt`, `.key` or `.csr` files and what they stand for or used to.